pypit/core/arextract.py

```diff
@@ -30,6 +30,934 @@
+    varimg = 1.0/(ivar + (ivar == 0.0))
+    var_box  = extract_boxcar(varimg*mask, specobj.trace_spat,box_radius, ycen = specobj.trace_spec)
+    nvar_box  = extract_boxcar(var_no*mask, specobj.trace_spat,box_radius, ycen = specobj.trace_spec)
+    sky_box  = extract_boxcar(skyimage*mask, specobj.trace_spat,box_radius, ycen = specobj.trace_spec)
+    rn2_box  = extract_boxcar(rn_img**2*mask, specobj.trace_spat,box_radius, ycen = specobj.trace_spec)
+    rn_posind = (rn2_box > 0.0)
+    rn_box = np.zeros(rn2_box.shape,dtype=float)
+    rn_box[rn_posind] = np.sqrt(rn2_box[rn_posind])
+    pixtot  = extract_boxcar(ivar*0 + 1.0, specobj.trace_spat,box_radius, ycen = specobj.trace_spec)
+    # If every pixel is masked then mask the boxcar extraction
+    mask_box = (extract_boxcar(ivar*mask == 0.0, specobj.trace_spat,box_radius, ycen = specobj.trace_spec) != pixtot)
+
+    bad_box = (wave_box <= 0.0) | (np.isfinite(wave_box) == False) | (box_denom == 0.0)
+    # interpolate bad wavelengths over masked pixels
+    if bad_box.any():
+        f_wave = RectBivariateSpline(spec_vec, spat_vec, waveimg)
+        wave_box[bad_box] = f_wave(specobj.trace_spec[bad_box], specobj.trace_spat[bad_box],grid=False)
+
+    ivar_box = 1.0/(var_box + (var_box == 0.0))
+    nivar_box = 1.0/(nvar_box + (nvar_box == 0.0))
+
+    specobj.boxcar['WAVE_BOX'] = wave_box
+    specobj.boxcar['FLUX_BOX'] = flux_box*mask_box
+    specobj.boxcar['IVAR_BOX'] = ivar_box*mask_box
+    specobj.boxcar['NIVAR_BOX'] = nivar_box*mask_box
+    specobj.boxcar['MASK_BOX'] = mask_box
+    specobj.boxcar['SKY_BOX'] = sky_box
+    specobj.boxcar['RN_BOX'] = rn_box
+
+    return
+
+
+
+
+def findfwhm(model, sig_x):
+    """ Calculate the spatial FWHM from an object profile. Utitlit routine for fit_profile
+
+    Parameters
+    ----------
+    model :   numpy float 2-d array [nspec, nspat]
+    x :
+
+    Returns
+    -------
+    peak :  Peak value of the profile model
+    peak_x:  sig_x location where the peak value is obtained
+    lwhm:   Value of sig_x at the left width at half maximum
+    rwhm:   Value of sig_x at the right width at half maximum
+
+    Revision History
+    ----------------
+    11-Mar-2005  Written by J. Hennawi and S. Burles David Schlegel, Princeton.
+    28-May-2018  Ported to python by J. Hennawi
+    """
+
+
+    peak = (model*(np.abs(sig_x) < 1.)).max()
+    peak_x = sig_x[(model*(np.abs(sig_x) < 1.)).argmax()]
+
+    lrev = ((sig_x < peak_x) & (model < 0.5*peak))[::-1]
+    lind, = np.where(lrev)
+    if(lind.size > 0):
+        lh = lind.min()
+        lwhm = (sig_x[::-1])[lh]
+    else:
+        lwhm = -0.5*2.3548
+
+    rind, = np.where((sig_x > peak_x) & (model < 0.5*peak))
+    if(rind.size > 0):
+        rh = rind.min()
+        rwhm = sig_x[rh]
+    else:
+        rwhm = 0.5 * 2.3548
+
+    return (peak, peak_x, lwhm, rwhm)
```

Comment:
The parentheses are superfluous.
The return is always a tuple when
it is multiple items.